src/createOauthClient.ts

```diff
@@ -4,6 +4,9 @@ import signIn from './signIn'
 import handleCallback from './handleCallback'
 import getAccessToken from './getAccessToken'
 
+const responseModes = <const>['fragment', 'query']
+export type ResponseMode = typeof responseModes[number]
```

Comment:
What is response mode used for?

Reply:
It is a setting to tell the issuer to include the callback parameters as query params or as query params in the location fragment (hash). 

So if you set response_mode=query you would get a callback like url this:

/redirecUri?code=xxx&state=yyy

And with response_mode=fragment you get

/redirecUri#code=xxx&state=yyy

An advantage of fragment is that code and state will not show up in access logs (not that it matters much since they are for one time use only).

The spec says that you should only provide a value for response_mode if you want another value than the default for your response_type so I changed from always setting fragment to allowing this to be set in config.

The reading on this is in a spec ammendment to oAuth in chapter 2 here: https://openid.net/specs/oauth-v2-multiple-response-types-1_0.html

Reply:
👍🏿